code/ui/generalhud/scoreboard/Scoreboard.ScoreboardGroup.cs

```diff
@@ -0,0 +1,118 @@
+        }
+
+        private class ScoreboardGroup : Panel
+        {
+            public readonly string GroupTitle;
+            public int GroupMembers = 0;
+            private readonly Panel _groupContent;
+            private readonly Panel _groupTitleWrapper;
+            private readonly Label _groupTitleLabel;
+
+            public ScoreboardGroup(Panel parent, string groupName)
+            {
+                Parent = parent;
+
+                GroupTitle = groupName;
+
+                AddClass(groupName);
+
+                _groupTitleWrapper = Add.Panel("scoreboardGroup__title-wrapper");
+                _groupTitleLabel = _groupTitleWrapper.Add.Label("", "scoreboardGroup__title");
+                _groupContent = Add.Panel("scoreboardGroup__content");
+            }
+
+            // TODO: Implement logic for the player counter in the title
+            public void UpdateLabel()
+            {
+                _groupTitleLabel.Text = $"{GroupTitle.ToUpper()}  -  {GroupMembers}";
+            }
+
+            public ScoreboardEntry AddEntry(PlayerScore.Entry entry)
+            {
+                ScoreboardEntry scoreboardEntry = _groupContent.AddChild<ScoreboardEntry>();
+                scoreboardEntry.ScoreboardGroupName = GroupTitle;
+                scoreboardEntry.SteamId = entry.Get<ulong>("steamid");
+
+                scoreboardEntry.UpdateFrom(entry);
+
+                return scoreboardEntry;
+            }
+        }
+
+        private ScoreboardGroup AddScoreboardGroup(string groupName)
+        {
+            if (_scoreboardGroups.ContainsKey(groupName))
+            {
+                return _scoreboardGroups[groupName];
+            }
+
+            ScoreboardGroup scoreboardGroup = new ScoreboardGroup(_mainContent, groupName);
+            scoreboardGroup.UpdateLabel();
+
+            _scoreboardGroups.Add(groupName, scoreboardGroup);
+
+            return scoreboardGroup;
+        }
+
+        private ScoreboardGroup GetScoreboardGroup(PlayerScore.Entry entry)
+        {
+            string group = DefaultScoreboardGroup.Alive.ToString();
+            ulong steamId = entry.Get<ulong>("steamid", 0);
+
+            if (steamId != 0)
+            {
+                foreach (Client client in Client.All)
+                {
+                    if (client.SteamId == steamId && client.Pawn != null)
+                    {
+                        if (client.Pawn is not TTTPlayer player)
+                        {
+                            break;
+                        }
+
+                        if (player.IsConfirmed)
+                        {
+                            group = DefaultScoreboardGroup.Dead.ToString();
+                        }
+                        else if (player.IsMissingInAction)
+                        {
+                            group = DefaultScoreboardGroup.MIA.ToString();
+                        }
+
+                        break;
+                    }
+                }
```

Comment:
Might be able to use our "GetAllPlayers" util method here.

Reply:
Uhm do you have a link to that method? I don't find anyone that would make sense to be integrated here sadly :/

Reply:
nvm, doesn't make sense in this case.